prusti-viper/src/encoder/foldunfold/mod.rs

```diff
@@ -1112,44 +985,7 @@ impl<'p, 'v: 'p, 'tcx: 'v> vir::CfgReplacer<PathCtxt<'p>, ActionVec> for FoldUnf
         &mut self,
         bcs: Vec<&PathCtxt<'p>>,
     ) -> Result<(Vec<ActionVec>, PathCtxt<'p>), Self::Error> {
-        trace!("[enter] prepend_join(..{})", &bcs.len());
-        assert!(!bcs.is_empty());
-        if bcs.len() == 1 {
-            Ok((vec![ActionVec(vec![])], bcs[0].clone()))
-        } else {
-            // Define two subgroups
-            let mid = bcs.len() / 2;
-            let left_pctxts = &bcs[..mid];
-            let right_pctxts = &bcs[mid..];
-
-            // Join the subgroups
-            let (left_actions_vec, mut left_pctxt) = self.prepend_join(left_pctxts.to_vec())?;
-            let (right_actions_vec, right_pctxt) = self.prepend_join(right_pctxts.to_vec())?;
-
-            // Join the recursive calls
-            let (merge_actions_left, merge_actions_right) = left_pctxt.join(right_pctxt)?;
-            let merged_pctxt = left_pctxt;
-
-            let mut branch_actions_vec: Vec<ActionVec> = vec![];
-            for mut left_actions in left_actions_vec {
-                left_actions.0.extend(merge_actions_left.iter().cloned());
-                branch_actions_vec.push(left_actions);
-            }
-            for mut right_actions in right_actions_vec {
-                right_actions.0.extend(merge_actions_right.iter().cloned());
-                branch_actions_vec.push(right_actions);
-            }
-
-            trace!(
-                "[exit] prepend_join(..{}): {}",
-                &bcs.len(),
-                branch_actions_vec
-                    .iter()
-                    .map(|v| format!("[{}]", v.iter().to_sorted_multiline_string()))
-                    .to_string()
-            );
-            Ok((branch_actions_vec, merged_pctxt))
-        }
+        prepend_join(bcs)
```

Comment:
Why should this `prepend_join` exist if it just delegates directly to the `prepend_join` below?

Reply:
Because `process_expiring_borrows.rs` needs to call `prepend_join`, but I didn't want to pass to it the mutable reference required to call this method. An alternative solution is to use `&self` instead of `&mut self` in the trait definition.